src/screens/Movie/Movie.tsx

```diff
@@ -55,13 +60,40 @@ const Movie = ({ match, location }: RouteComponentProps<MovieRouteParams>): JSX.
   const { data: playlist } = usePlaylist(features?.recommendationsPlaylist || '', { related_media_id: id });
 
   // Favorite
-  const isFavorited = useFavoritesStore((state) => !!item && state.hasItem(item));
+  const { isFavorited, favoritesCount } = useFavoritesStore((state) => ({
+    isFavorited: !!item && state.hasItem(item),
+    favoritesCount: state.favorites?.length || 0,
+  }));
 
   // User, entitlement
   const { user, subscription } = useAccountStore(({ user, subscription }) => ({ user, subscription }), shallow);
   const { isEntitled } = useEntitlement(item);
 
   // Handlers
+  const onFavoritesWarningToggle = useCallback(() => {
+    setIsFavoritesWarningShown(!isFavoritesWarningShown);
+  }, [setIsFavoritesWarningShown, isFavoritesWarningShown]);
+
+  const onFavoriteButtonClick = useCallback(() => {
+    if (!item) {
+      return;
+    }
+
+    if (isFavorited) {
```

Comment:
Did you try clicking the favorite button really fast to make sure this isn't a race condition? It seems like it should be ok, but I can't quite mentally map the flow with the callback, the store access, and the local variable.

Reply:
I tired it know. Works fine.